tests/tizen_capi/unittest_tizen_capi.cc

```diff
@@ -4988,6 +4988,113 @@ TEST (nnstreamer_capi_singleshot, invoke_dynamic_fail_n)
   g_free (test_model);
 }
 
+/**
+ * @brief Test for internal function 'ml_validate_model_file'.
+ * @detail Invalid params.
+ */
+TEST (nnstreamer_capi_internal, validate_model_file_01_n)
+{
+  const gchar cf_name[] = "libnnstreamer_customfilter_passthrough_variable" \
+      NNSTREAMER_SO_FILE_EXTENSION;
+  int status;
+  ml_nnfw_type_e nnfw = ML_NNFW_TYPE_CUSTOM_FILTER;
+  const gchar *root_path = g_getenv ("NNSTREAMER_BUILD_ROOT_PATH");
+  gchar *test_model;
+
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  test_model = g_build_filename (root_path, "build", "nnstreamer_example",
+      "custom_example_passthrough", cf_name, NULL);
+  ASSERT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  status = ml_validate_model_file (NULL, 1, &nnfw);
+  EXPECT_NE (status, ML_ERROR_NONE);
+
+  status = ml_validate_model_file (&test_model, 0, &nnfw);
+  EXPECT_NE (status, ML_ERROR_NONE);
+
+  status = ml_validate_model_file (&test_model, 1, NULL);
+  EXPECT_NE (status, ML_ERROR_NONE);
+
+  g_free (test_model);
+}
+
+/**
+ * @brief Test for internal function 'ml_validate_model_file'.
+ * @detail Invalid file extension.
+ */
+TEST (nnstreamer_capi_internal, validate_model_file_02_n)
+{
+  const gchar cf_name[] = "libnnstreamer_customfilter_passthrough_variable" \
+      NNSTREAMER_SO_FILE_EXTENSION;
+  int status;
+  ml_nnfw_type_e nnfw;
+  const gchar *root_path = g_getenv ("NNSTREAMER_BUILD_ROOT_PATH");
+  gchar *test_model1, *test_model2;
+  gchar *test_models[2];
+
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  test_model1 = g_build_filename (root_path, "build", "nnstreamer_example",
+      "custom_example_passthrough", cf_name, NULL);
+  test_model2 = g_build_filename (root_path, "tests", "test_models", "models",
+      "mobilenet_v1_1.0_224_quant.tflite", NULL);
+  ASSERT_TRUE (g_file_test (test_model1, G_FILE_TEST_EXISTS));
+  ASSERT_TRUE (g_file_test (test_model2, G_FILE_TEST_EXISTS));
```

Comment:
This line seems to have potential memory leak.

Reply:
Right. If svace detects these lines, let's ignore that. (false-positive)